source/material_model/visco_plastic.cc

```diff
@@ -70,60 +70,31 @@ namespace aspect
       return cohesions;
     }
 
+
+
     template <int dim>
-    std::vector<double>
+    double
     ViscoPlastic<dim>::
-    compute_volume_fractions( const std::vector<double> &compositional_fields) const
+    average_value ( const std::vector<double> &composition,
+                    const std::vector<double> &parameter_values,
+                    const enum averaging_scheme &average_type) const
     {
-      std::vector<double> volume_fractions( compositional_fields.size()+1);
-
-      // clip the compositional fields so they are between zero and one
-      std::vector<double> x_comp = compositional_fields;
-      for ( unsigned int i=0; i < x_comp.size(); ++i)
-        x_comp[i] = std::min(std::max(x_comp[i], 0.0), 1.0);
+      double averaged_parameter = 0.0;
 
-      // assign compositional fields associated with strain a value of 0
+      // Store which components to exclude during volume fraction computation.
+      ComponentMask composition_mask(this->n_compositional_fields(),true);
       if (use_strain_weakening == true)
```

Comment:
What if the strain tensor is tracked, but not used for weakening? Is that not supported? (I know that technically this doesn't relate to your pull request...)

Reply:
The `use_finite_strain_tensor` variable in this material model explicitly means use it for strain weakening. It is not meant for anything else.